api/namex/services/nro/utils.py

```diff
@@ -41,3 +40,68 @@ def validNRFormat(nr):
         return False
 
     return True
+
+
+def generate_compressed_name(original_name):
+
+    # Removing all instances of "THE " and " THE "; no need to removed " THE".
+    def _delete_the(in_name):
+        out_name = in_name
+        if len(in_name) > 4:
+            if in_name[:4] == "THE ":
+                out_name = in_name[4:]
+
+            out_name = out_name.replace(" THE ", "")
+
+        return out_name
+
+    def _remove_char(in_name):
+        chars = set('ABCDEFGHIJKLMNOPQRSTUVWXYZ#&0123456789')
+
+        out_name = in_name
+
+        for c in out_name:
+            if not (c in chars):
+                out_name = out_name.replace(c, "")
+
+        return out_name
+
+    def _translate_char(in_name):
+        out_name = in_name.replace('&', 'AND')
+        out_name = out_name.replace('#', 'NUMBER')
+        out_name = out_name.replace('1', 'ONE')
+        out_name = out_name.replace('2', 'TWO')
+        out_name = out_name.replace('3', 'THREE')
+        out_name = out_name.replace('4', 'FOUR')
+        out_name = out_name.replace('5', 'FIVE')
+        out_name = out_name.replace('6', 'SIX')
+        out_name = out_name.replace('7', 'SEVEN')
+        out_name = out_name.replace('8', 'EIGHT')
+        out_name = out_name.replace('9', 'NINE')
+        out_name = out_name.replace('0', 'ZERO')
+        return out_name
+
+    def _translate_bc(in_name):
+        out_name = in_name
+        if len(in_name) > 15:
+            if "BRITISHCOLUMBIA" in in_name[:15]:
```

Comment:
`if in_name.startswith("BRITISHCOLUMBIA"):`
Consider using startswith instead of the 2 if statements